test/suite/test_timestamp22.py

```diff
@@ -0,0 +1,444 @@
+        # here and we have multiple messages to sort out.
+        self.checkStderr()
+
+        # 'yield' runs the subordinate operation, we'll catch any resulting exceptions.
+        try:
+            if expected == self.FAILURE:
+                # Soak up any error messages that happen as a result of the failure.
+                with self.expectedStderrPattern(r'^.*$', re_flags=re.MULTILINE):
+                    yield
+            else:
+                yield
+            got = self.SUCCESS
+        except:
+            got = self.FAILURE
+            self.cleanStderr()
+
+        message += ' got ' + got
+
+        # If we're about to assert, show some extra info
+        if expected != got:
+            message += ': ERROR expected ' + expected
+            self.checkStderr()
+        self.pr(message)
+        self.assertEquals(expected, got)
+
+    # Create a predictable value based on the iteration number and timestamp.
+    def gen_value(self, iternum, ts):
+        return str(iternum) + '_' + str(ts) + '_' + 'x' * 1000
+
+    # Given a number representing an "approximate timestamp", generate a timestamp
+    # that is near that number, either plus or minus.
+    def gen_ts(self, approx_ts):
+        # a number between -10 and 10:
+        n = self.rand.rand32() % 21 - 10
+        ts = approx_ts + n
+        if ts <= 0:
+            ts = 1
+        return ts
+
+    # Asks whether we should do an illegal operation now. Return yes 5%.
+    def do_illegal(self):
+        return self.rand.rand32() % 20 == 0
+
+    def report(self, func, arg = None):
+        self.pr('DOING: ' + func + ('' if arg == None else '(' + arg + ')'))
+
+    # Insert a set of rows, each insert in its own transaction, with the
+    # given timestamps.
+    def updates(self, value, ds, do_prepare, commit_ts, durable_ts, read_ts):
+
+        # Generate a configuration for a timestamp_transaction() call.
+        # Returns: 1) whether it expects success, 2) config 3) new running commit timestamp
+        def timestamp_txn_config(commit_ts, running_commit_ts):
+            ok = True
+            config = ''
+            this_commit_ts = -1
+            if self.do_illegal():
+                # setting durable timestamp must be after prepare call
+                config += ',durable_timestamp=' + timestamp_str(self.gen_ts(commit_ts))
+                ok = False
+
+            # ODDITY: if we set the durable timestamp (which is illegal at this point), and set a
+            # valid commit timestamp, the timestamp_transaction() call will fail, but apparently,
+            # only the durable part fails.  The evidence is that the commit timestamp is set,
+            # as we get a complaint to that effect at the prepare call.  The issue is described
+            # in WT-6995.  This seems wrong, and it's hard to work around the problem, so we'll just
+            # avoid testing that situation for now.  Hence the check for a blank configuration.
```

Comment:
Minor: I think the comment should have an explicit statement about what change to make once WT-6995 is fixed such as "Once WT-6995 is fixed, remove the entire clause `and config == ''`. And perhaps add a sentence to that ticket reminding the person working on it that this part specifically needs fixing (if it isn't already).